pghoard/common.py

```diff
@@ -353,5 +354,25 @@ def increase_pipe_capacity(*pipes):
                 pass
 
 
+class PGHoardThread(Thread):
+    def __init__(self):
+        super().__init__()
+        self.exception: Optional[Exception] = None
+
+    def run_safe(self):
+        raise NotImplementedError()
+
+    def run(self):
+        try:
+            self.run_safe()
+        except Exception as ex:  # pylint: disable=broad-except
+            self.exception = ex
+
+    def join(self, timeout=None):
+        super().join(timeout)
+        if self.exception:
+            raise self.exception
```

Comment:
just a question/curiosity: Why this form and not `raise WhateverException("...") from self.exception` so that both tracebacks are visible?

Reply:
Done !